### PR TITLE
[ROCm] Enabling unit test fft_ops_test.py on ROCm

### PR DESCRIPTION
**Pull request by deven-amd**
This PR is based on the original changes made by @ekuznetsov139 here - https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/1167

----------------------

Fix for a rocFFT bug in real<-->complex transforms

The input buffer contents can be over-written by rocFFT for all of the real <---> complex transforms (R2C, C2R, D2Z, Z2D), even when transform explicitly provides a separate output buffer.

See following issues for more details

* ROCmSoftwarePlatform/rocFFT#298
* ROCmSoftwarePlatform#1150

This commit adds a change to copy the input buffer (for the above transforms) and use the input copy when calling the rocFFT api to do the transform

--------------------

 Updating fft_ops_test.py for ROCm

1. Some testcases were being skipped on the ROCm platform. Those tests are now passing and hence being enabled by this commit

2. For testcases that test the irfft transform, the input needs to be massaged on the ROCm platform. The ROCm irfft transform expects its input to be in the form that is output by the rfft transform.

---------------------

/cc @cheshire @chsigg @ekuznetsov139

### CODE REVIEW

**Comment by deven-amd** [comment]
@cheshire @chsigg gentle ping

File: tensorflow/python/kernel_tests/signal/fft_ops_test.py [hunk 1]
```diff
@@ -457,6 +460,8 @@ def test_fft_lenth_truncate(self, rank, extra_dims, size, np_rtype):
       VALID_FFT_RANKS, range(3), (5, 6), (np.float32, np.float64)))
   def test_fft_lenth_pad(self, rank, extra_dims, size, np_rtype):
     """Test padding (FFT size > dimensions)."""
+    # if test.is_built_with_rocm() and size == 6:
```
Comment (cheshire) [inline tensorflow/python/kernel_tests/signal/fft_ops_test.py:463]: Remove comment?

**Review by cheshire — APPROVED** [review]

**Comment by gbaned** [comment]
@deven-amd  Can you please check @cheshire's comments and keep us posted ? Thanks!

File: tensorflow/python/kernel_tests/signal/fft_ops_test.py (hunk 1 above)
Comment (deven-amd) [reply tensorflow/python/kernel_tests/signal/fft_ops_test.py]: done

**Comment by deven-amd** [comment]
@gbaned I have addressed the feedback from @cheshire and rebased to tip.

**Review by cheshire — APPROVED** [review]

**Pull request merged**